### PR TITLE
Update episode03_configuration

### LINKED ISSUE #11: Episode 03 (Configuration)
I started working on episode 03 in the current structure that is about configuration. Also, I made a draft pull request for that. 

Checklist:

- [x] make a draft version
- [x] update questions, objectives and key points
- [x] add the content
- [x] add challenges to the episode (if it is needed).

### PR DESCRIPTION
**Pull request by SarahAlidoost**
# Pull Request checklist

We appreciate your time and effort to improve the tutorial. Please keep in mind that lesson maintainers are volunteers and it may be some time before they can respond to your contribution.

* * *

## Before you start

- [x] Read [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md).
- [x] Create an [issue](https://github.com/ESMValGroup/tutorial/issues) to discuss your idea. This allows your contributions to be incorporated into the tutorial.

## Tasks

- [x] Give this pull request a descriptive title.
- [x] If you are contributing to existing lesson materials, please make sure the content conforms to the `Lesson development` section in [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md) and does not contain any spelling or grammatical errors.
- [x] If you are making a new episode, please make sure the content conforms to the `Lesson organization` and `Lesson formatting` sections in [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md) and does not contain any spelling or grammatical errors.
- [ ] Preferably Codacy checks pass. Status can be seen below your pull request. If there is an error, click the link to find out why.
- [x] Preview changes on your machine before pushing them to GitHub by running `make serve`, alternatively `make docker-serve`. Please see the `Previewing your changes locally` section in [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md) for installation instructions.
- [x] All code instructions have been tested.

If you need help with any of the tasks above, please do not hesitate to ask by commenting in the issue or pull request.

* * *

Closes  #11

### CODE REVIEW

File: _episodes/03-configuration.md [hunk 1]
```diff
@@ -3,91 +3,158 @@ title: "Configuration"
...
 The ``config-user.yml`` configuration file contains all the global level information needed by ESMValTool to run.
-This is an (YAML file) [https://yaml.org/spec/1.2/spec.html]. An example configuration file can be found in the root directory of the ESMValTool repository.
-Make a copy and rename it to ``config-user.yml``:
+This is an [YAML file](https://yaml.org/spec/1.2/spec.html). An example configuration file can be found in the root directory of the ESMValTool repository.
+Let's change our working directory to ESMValTool, then make a copy of the file and rename it to ``config-user.yml``, then run a text editor called Nano to open it:
 
-~~~
+~~~bash
+  cd ESMValTool
   cp config-user-example.yml config-user.yml
```
Comment (ledm) [inline _episodes/03-configuration.md]: My workflow is to copy this to a new directory next to ESMValTool called run, or workspace. I find this works better in keeping things in one place. 

Is there a best practices document somewhere that we should consult?

File: _episodes/03-configuration.md [hunk 2]
```diff
@@ -3,91 +3,158 @@ title: "Configuration"
...
+  OBS6: ~/obs6_inputpath
+  obs4mips: ~/obs4mips_inputpath
+  ana4mips: ~/ana4mips_inputpath
+  native6:  ~/native6_inputpath
+  RAWOBS: ~/rawobs_inputpath
+  default: ~/default_inputpath
 ~~~
+
+In this lesson, we will work with data from [CMIP5](https://esgf-node.llnl.gov/projects/cmip5/).
+Add the root path of the folder where you downloaded the data during the [Setup](https://esmvalgroup.github.io/tutorial/setup.html).
```
Comment (ledm) [inline _episodes/03-configuration.md]: This assumes that people will be working locally, instead of somewhere like Jasmin, where the data will be available already. 

In the long term, a local installation is useful, but which version do we want to teach in the tutorial? Local or Jasmin? @valeriupredoi @bouweandela , thoughts?

Alternatively, do we want to write this documentation twice? Once for local and once for jasmin?

Comment (bouweandela) [reply _episodes/03-configuration.md]: Maybe we can make the tutorial for a local installation and add commented out versions of the correct drs settings to config-user-example.yml, so people can just uncomment the lines they need if they happen to be on a cluster that is connected to an ESGF node?

File: _episodes/03-configuration.md (hunk 1 above)
Comment (SarahAlidoost) [reply _episodes/03-configuration.md]: @ledm Thanks. I like your suggestion. I couldn't find any best practices for this issue. However, I have checked some lessons. There is no standard way, some choose "code", "data" directories in Desktop which is not good. I applied this workflow: 

```
- esmvaltool_tutorial
   - ESMValTool (Github repo)
   - data
   - other files like a copy of config file. 
```

Instruction on how to make these directories will come in `setup` section of the tutorial.

File: _episodes/03-configuration.md (hunk 2 above)
Comment (SarahAlidoost) [reply _episodes/03-configuration.md]: @ledm @bouweandela I agree with Bouwe to make the tutorial for local use. I updated this part of the lesson. Now the instruction on how to get the data (or its rootpath) will come in `setup` section of the tutorial. This is the most common approach for several lessons that I have seen in software carpentry.  The `setup` section contains everything related to data and additional material. Therefore, the episode content won't be changed.

Comment (bouweandela) [reply _episodes/03-configuration.md]: Created an issue to do this: https://github.com/ESMValGroup/ESMValTool/issues/1622

File: _episodes/03-configuration.md (hunk 1 above)
Comment (bouweandela) [reply _episodes/03-configuration.md]: There is an open issue about making config-user.yml more user friendly: https://github.com/ESMValGroup/ESMValCore/issues/64

**Comment by bouweandela** [comment]
@ledm Are you OK with the changes? Can this be merged?

File: _episodes/03-configuration.md [hunk 3]
```diff
@@ -3,91 +3,178 @@ title: "Configuration"
 teaching: 0
 exercises: 0
 questions:
-- "What is user configuration file and how can I use it?"
+- "What is the user configuration file and how can I use it?"
 objectives:
```
Comment (hb326) [inline _episodes/03-configuration.md:7]: I would probably rephrase this question to "What is the user configuration file and how do I use it?" Otherwise it sounds a little too optional for me, like it would actually be possible to ignore the user configuration file and run the ESMValTool without it.

File: _episodes/03-configuration.md [hunk 4]
```diff
@@ -3,91 +3,178 @@ title: "Configuration"
...
-- "Understand the data directories structure"
+- "Understand the contents of the user-config.yml file"
+- "Prepare the user-config.yml file for use"
```
Comment (hb326) [inline _episodes/03-configuration.md:9]: I would probable rephrase this point here: "Prepare a personalized user-config.yml file" or something like it?

File: _episodes/03-configuration.md [hunk 5]
```diff
@@ -3,91 +3,178 @@ title: "Configuration"
...
 - "Configure ESMValTool to ignore some settings"
```
Comment (hb326) [inline _episodes/03-configuration.md:10]: I am not sure I understand what you mean here. You want to set settings in the user-config.yml file so that the ESMValTool is ignoring some things it would normally do?

File: _episodes/03-configuration.md [hunk 6]
```diff
@@ -3,91 +3,178 @@ title: "Configuration"
...
 keypoints:
-- "The ``config-user.yml`` file tells ESMValTool what data are input"
-- "The ``config-user.yml`` file tells ESMValTool what directory is the destination"
+- "The ``config-user.yml`` tells ESMValTool where to find input data."
+- " ``rootpath`` determines root directory for input data."
+- " ``output_dir`` is the destination directory."
```
Comment (hb326) [inline _episodes/03-configuration.md:14]: Maybe rephrase to "``output_dir`` defines the destination directory."?

File: _episodes/03-configuration.md [hunk 7]
```diff
@@ -3,91 +3,178 @@ title: "Configuration"
...
 objectives:
-- "Understand the data directories structure"
+- "Understand the contents of the user-config.yml file"
+- "Prepare the user-config.yml file for use"
 - "Configure ESMValTool to ignore some settings"
 keypoints:
-- "The ``config-user.yml`` file tells ESMValTool what data are input"
-- "The ``config-user.yml`` file tells ESMValTool what directory is the destination"
+- "The ``config-user.yml`` tells ESMValTool where to find input data."
+- " ``rootpath`` determines root directory for input data."
```
Comment (hb326) [inline _episodes/03-configuration.md:13]: Maybe rephrase to "``rootpath`` defines the root directory for the input data."

File: _episodes/03-configuration.md [hunk 8]
```diff
@@ -3,91 +3,178 @@ title: "Configuration"
...
 ---
 
 ## The configuration file
 
 The ``config-user.yml`` configuration file contains all the global level information needed by ESMValTool to run.
-This is an (YAML file) [https://yaml.org/spec/1.2/spec.html]. An example configuration file can be found in the root directory of the ESMValTool repository.
-Make a copy and rename it to ``config-user.yml``:
+This is an [YAML file](https://yaml.org/spec/1.2/spec.html). An example configuration file can be found in the root directory of the ESMValTool repository: [config-user-example.yml](https://github.com/ESMValGroup/ESMValTool/blob/master/config-user-example.yml).
 
+Let's download it to our working directory ``esmvaltool_tutorial`` that is made during the [Setup](https://esmvalgroup.github.io/tutorial/setup.html).
```
Comment (hb326) [inline _episodes/03-configuration.md:22]: Maybe rephrase to "... that was created during the [Setup]..."?

File: _episodes/03-configuration.md [hunk 9]
```diff
@@ -3,91 +3,178 @@ title: "Configuration"
...
+To do that, click on [this link](https://raw.githubusercontent.com/ESMValGroup/ESMValTool/master/config-user-example.yml) to see a raw version of the file, right-click and press ``save as``, then you can rename it to ``config-user.yml`` and save it into the working directory ``esmvaltool_tutorial``.
+
+Now in a terminal, let's change our working directory to ``esmvaltool_tutorial``. Then, we run a text editor called Nano to open the configuration file:
```
Comment (hb326) [inline _episodes/03-configuration.md:25]: Maybe rephrase to "Now, let's change our working directory in a terminal window to ..."

File: _episodes/03-configuration.md [hunk 10]
```diff
@@ -3,91 +3,178 @@ title: "Configuration"
...
-
-## Rootpaths to input data
-ESMValTool uses several categories (in ESMValTool, this is referred to as projects) for input data based on their source, like CMIP for dataset from climate model intercomparison project, and OBS for observational dataset that adhere to (CMOR standard)[https://cmor.llnl.gov/].
-For each category, you can define either one path or several pathes as a list.
-In this lesson, you work with data from (CMIP5)[https://esgf-node.llnl.gov/projects/cmip5/].
-Add the root path of the folder where you downloaded the data during the (Setup)[https://escience-academy.github.io/lesson-esmvaltool/setup.html].
 
+* Rootpath to input data
+* Directory structure for the data from different projects
+* Number of parallel tasks
```
Comment (hb326) [inline _episodes/03-configuration.md:36]: Maybe rephrase to "* Number of tasks that can be run in parallel"

File: _episodes/03-configuration.md [hunk 11]
```diff
@@ -3,91 +3,178 @@ title: "Configuration"
...
+* Destination directory
+* Auxiliary data directory
+* Output settings
+
+> ## Which text editor
```
Comment (hb326) [inline _episodes/03-configuration.md:41]: I am not sure why you called this section "Which text editor". With this header I would have expected some text about the pros and cons of different text editors. Maybe rephrase the header to "Text editor side note" or something

File: _episodes/03-configuration.md [hunk 12]
```diff
@@ -3,91 +3,178 @@ title: "Configuration"
...
+>
+> No matter what editor you use, you will need to know where it searches for and saves files. If you start it from the shell, it will (probably) use your current working directory as its default location. We use ``nano`` in examples because it is one of the least complex text editors. Press <kbd>ctrl</kbd> + <kbd>O</kbd> to save the file, and then <kbd>ctrl</kbd> + <kbd>X</kbd> to exit ``nano``.
```
Comment (hb326) [inline _episodes/03-configuration.md:43]: I would probably rephrase to "... We use ``nano`` in examples here because...".

File: _episodes/03-configuration.md [hunk 13]
```diff
@@ -3,91 +3,178 @@ title: "Configuration"
...
+{: .callout}
```
Comment (hb326) [inline _episodes/03-configuration.md:44]: This command is still visible in the text in it's formatted form. Shouldn't be, right?

File: _episodes/03-configuration.md [hunk 14]
```diff
@@ -3,91 +3,178 @@ title: "Configuration"
...
+
+## Rootpath to input data
+
+ESMValTool uses several categories (in ESMValTool, this is referred to as projects) for input data based on their source.
+The current categories in the configuration file are mentioned below. For example, CMIP is used for a dataset from the climate model intercomparison project whereas OBS for an observational dataset. The ``rootpath`` specifies the directories where ESMValTool will look for input data. For each category, you can define either one path or several paths as a list.
```
Comment (hb326) [inline _episodes/03-configuration.md:49]: I would change the wording to "... CMIP is used for a dataset from the Climate Model Intercomparison Project whereas OBS is used for an observational dataset."

File: _episodes/03-configuration.md [hunk 15]
```diff
@@ -3,91 +3,178 @@ title: "Configuration"
...
+  CMIP3: [~/cmip3_inputpath1, ~/cmip3_inputpath2]
+  CMIP5: [~/cmip5_inputpath1, ~/cmip5_inputpath2]
+  CMIP6: [~/cmip6_inputpath1, ~/cmip6_inputpath2]
+  OBS: ~/obs_inputpath
+  OBS6: ~/obs6_inputpath
+  obs4mips: ~/obs4mips_inputpath
+  ana4mips: ~/ana4mips_inputpath
+  native6:  ~/native6_inputpath
+  RAWOBS: ~/rawobs_inputpath
+  default: ~/default_inputpath
```
Comment (hb326) [inline _episodes/03-configuration.md:62]: Do we want to add an explanation about the other projects mentioned here beside CMIP and OBS? Or do we want to refer to a different episode where these projects might be described (like how to add your own observational dataset)?

File: _episodes/03-configuration.md [hunk 16]
```diff
@@ -3,91 +3,178 @@ title: "Configuration"
...
 ~~~
+
+In this lesson, we will work with data from [CMIP5](https://esgf-node.llnl.gov/projects/cmip5/).
+We add the root path of the folder where data is available.
```
Comment (hb326) [inline _episodes/03-configuration.md:66]: Maybe rephrase to "We add the root path of the folder where our/your data is available." I am not sure which one of the pronouns I would prefer, but I think it would clarify that the path to the student's data should be added here.

File: _episodes/03-configuration.md [hunk 17]
```diff
@@ -3,91 +3,178 @@ title: "Configuration"
...
+    CMIP5: [~/cmip5_inputpath1, ~/cmip5_inputpath2, ~/esmvaltool_tutorial/data]
 ~~~
-{: .source}
 
-## Auxiliary data directory (used for some additional datasets)
-  auxiliary_data_dir: ~/auxiliary_data
+> ## Setting the correct rootpath
+>
+> * To get the data (or its correct rootpath), check instruction in [Setup](https://esmvalgroup.github.io/tutorial/setup.html).
+> * For more information about setting the rootpath, you can visit ESMValTool [documentation](https://esmvaltool.readthedocs.io/projects/esmvalcore/en/latest/esmvalcore/datafinder.html).
```
Comment (hb326) [inline _episodes/03-configuration.md:77]: I would rephrase to "For more informatino about setting the roothpath, see also ESMValTool...".

File: _episodes/03-configuration.md [hunk 18]
```diff
@@ -3,91 +3,178 @@ title: "Configuration"
...
-    CMIP5: [~/cmip5_inputpath1, ~/cmip5_inputpath2, ~/escience-academy/test_data]
+    CMIP5: [~/cmip5_inputpath1, ~/cmip5_inputpath2, ~/esmvaltool_tutorial/data]
 ~~~
-{: .source}
 
-## Auxiliary data directory (used for some additional datasets)
-  auxiliary_data_dir: ~/auxiliary_data
+> ## Setting the correct rootpath
+>
+> * To get the data (or its correct rootpath), check instruction in [Setup](https://esmvalgroup.github.io/tutorial/setup.html).
```
Comment (hb326) [inline _episodes/03-configuration.md:76]: I am not sure what you mean here. Is this a pointer toward instructions on where to actually get the CMIP5 data or is this a pointer toward the correct setup of the rootpath?

File: _episodes/03-configuration.md [hunk 19]
```diff
@@ -3,91 +3,178 @@ title: "Configuration"
...
+> * For more information about setting the rootpath, you can visit ESMValTool [documentation](https://esmvaltool.readthedocs.io/projects/esmvalcore/en/latest/esmvalcore/datafinder.html).
+{: .callout}
```
Comment (hb326) [inline _episodes/03-configuration.md:78]: This command is also visible in the formatted text.

File: _episodes/03-configuration.md (hunk 8 above)
Comment (bouweandela) [reply _episodes/03-configuration.md:22]: It might be good to create the link as described in the section [Locations and Names](https://carpentries.github.io/lesson-example/04-formatting/index.html), because then it will still work even if the repository is moved/renamed.

File: _episodes/03-configuration.md [hunk 20]
```diff
@@ -3,91 +3,178 @@ title: "Configuration"
...
-Python toolkits such as cartopy will attempt to download data files at run
-time, typically geographic data files such as coastlines or land surface maps.
-This can fail if the machine does not have access to the wider internet. This
-location allows us to tell cartopy (and other similar tools) where to find the
-files if they can not be downloaded at runtime. To reiterate, this setting is
-not for model or observational datasets, rather it is for data files used in
-plotting such as coastline descriptions and so on.
+## Directory structure for the data from different projects
 
+Input data can be from various models, observations and reanalysis data that adhere to the [CF/CMOR standard](https://cmor.llnl.gov/). To set a directory, you can use one of the values of ``default``, ``BADC``, ``DKRZ``, ``ETHZ``, .... Let's use ``default`` in our example:
```
Comment (hb326) [inline _episodes/03-configuration.md:83]: I think we might have to mention here that the information given here for "drs" refers to a folder structure that describes how the data is stored. Or would that be too much information for beginners here? If we don't mention it, I think it is not clear to the students what the difference between "BADC" and "ETHZ" would mean.

File: _episodes/03-configuration.md [hunk 21]
```diff
@@ -3,91 +3,178 @@ title: "Configuration"
...
+
+> ## Available drs
+>
+> For more information about directories, you can visit ESMValTool [documentation](https://esmvaltool.readthedocs.io/projects/esmvalcore/en/latest/esmvalcore/config.html#developer-configuration-file).
+{: .callout}
+
+## Number of parallel tasks
 
-.. code-block:: yaml
+This option enables you to perform parallel processing. You can choose the number of tasks in parallel as 1/2/3/4/... or you can set it to ``null`` that tells ESMValTool to use the number of available CPUs:
```
Comment (hb326) [inline _episodes/03-configuration.md:96]: I would rephrase to "... or you can set it to ``null``. That tells ESMValTool...".

Comment (hb326) [inline _episodes/03-configuration.md:96]: Does it tell the ESMValTool to use the maximum number of available GPUs if it is set to ``null``?

File: _episodes/03-configuration.md [hunk 22]
```diff
@@ -3,91 +3,178 @@ title: "Configuration"
...
-turns on or off the writing of netCDF files.
+max_parallel_tasks: null
+~~~
 
-The ```rootpath`` specifies the directories where ESMValTool will look for input
-data. Similarly, ``output_dir`` specifies where ESMValTool will store its
-output, i.e. figures, data, logs, etc. Make sure to set appropriate paths.
+> ## Set the number of tasks
+>
+> If you run out of memory, try setting ``max_parallel_tasks`` to 1. Then, check the amount of memory you need for that by inspecting the file ``run/resource_usage.txt`` in the output directory. Using the number there you can increase the number of parallel tasks again to a reasonable number for the amount of memory available in your system.
```
Comment (hb326) [inline _episodes/03-configuration.md:105]: This is probably also something that we should link to the "debugging" episode.

File: _episodes/03-configuration.md [hunk 23]
```diff
@@ -3,91 +3,178 @@ title: "Configuration"
...
+{: .callout}
 
-.. code-block:: yaml
+## Destination directory
 
+The destination directory is the rootpath where ESMValTool will store its output, i.e. figures, data, logs, etc. With every run, ESMValTool automatically generates a new output folder determined by the recipe name, and date and time using the format: YYYYMMDD_HHMMSS. This folder contains four further subfolders: ``plots``, ``preproc``, ``run``, ``work``.
```
Comment (hb326) [inline _episodes/03-configuration.md:110]: I would rephrase to "... by recipe name, date and time ..."

File: _episodes/03-configuration.md [hunk 24]
```diff
@@ -3,91 +3,178 @@ title: "Configuration"
...
 
+Let's name our destination directory as ``esmvaltool_output`` in the working directory:
```
Comment (hb326) [inline _episodes/03-configuration.md:112]: I would rephrase to "... our destinateion directory ``esmvaltool_output`` in ..."

File: _episodes/03-configuration.md [hunk 25]
```diff
@@ -3,91 +3,178 @@ title: "Configuration"
...
+output_dir: ./esmvaltool_output
+~~~
 
-You can tailor it for your system using the explanation below.
+> ## Content of subfolders
+>
+> * ``plots``: the location for all plots, split by individual diagnostics and fields.
+> * ``preproc``: this folder contains all the preprocessed data and metadata.yml interface files. Note that by default this directory will be deleted after each run because most users will only need the results from the diagnostic scripts.
+> * ``run``: this folder includes all log files, a copy of the recipe, a summary of the resource usage, and the settings.yml interface files, resource_usage.txt and temporary files created by the diagnostic scripts.
+> * ``work``: a place for any diagnostic script results that are not plots, e.g. files in NetCDF format (depends on the diagnostic script).
```
Comment (hb326) [inline _episodes/03-configuration.md:123]: I would rephrase to be consistent with the entries before "``work``: this folder is a place for any..."

File: _episodes/03-configuration.md [hunk 26]
```diff
@@ -3,91 +3,178 @@ title: "Configuration"
...
+> We explain more about output in the next [lesson](https://esmvalgroup.github.io/tutorial/04-toy-example/index.html)
+{: .callout}
 
-.. note::
+## Auxiliary data directory
 
-   The ``config-user.yml`` file is specified as argument at run time, so it is
-   possible to have several available with different purposes: one for
-   formalised runs, one for debugging, etc...
+The ``auxiliary_data_dir`` setting is the path to place any required additional auxiliary data files. This location allows us to tell the diagnostic script where to find the files if they can not be downloaded at runtime. This option is not for model or observational datasets, rather it is for data files used in plotting such as coastline descriptions and so on.
```
Comment (hb326) [inline _episodes/03-configuration.md:130]: I would rephrase "... is the path where any additionally required auxiliary data files are stored." And "This option should not be used for model or observational datasets, but for data files (e.g. shape files) used in plotting, such as ...".

File: _episodes/03-configuration.md [hunk 27]
```diff
@@ -3,91 +3,178 @@ title: "Configuration"
...
 
+~~~YAML
+auxiliary_data_dir: ~/auxiliary_data
+~~~
 
-{% include links.md %}
+## Output settings
 
+These settings are used to inform ESMValTool about your preference. You can turn on or off the setting by ``true`` or ``false`` values. Most of these settings are fairly self-explanatory, ie:
```
Comment (hb326) [inline _episodes/03-configuration.md:138]: Maybe rephrase to "These settings are used to inform ESMValTool about your preference about specific actions. "

File: _episodes/03-configuration.md [hunk 28]
```diff
@@ -3,91 +3,178 @@ title: "Configuration"
...
+# Only available for Python diagnostics
+profile_diagnostic: false
+~~~
+
+> ## Make your own configuration file
+>
+> It is possible to have several configuration files with different purposes, for example: config-user_formalised_runs.yml, config-user_debugging.yml
+{: .callout}
+
+> ## Different settings
```
Comment (hb326) [inline _episodes/03-configuration.md:170]: Is this already one of the tasks for people to try out? It is not clear to me from the placing...

File: _episodes/03-configuration.md [hunk 29]
```diff
@@ -3,91 +3,178 @@ title: "Configuration"
...
+>
+> In the configuration file, which settings are useful to store preprocessed data?
```
Comment (hb326) [inline _episodes/03-configuration.md:172]: I would probably rephrase to "In the configuration file, which settings are usefule to make sure preprocessed data is stored when ESMValTool is run?

File: _episodes/03-configuration.md (hunk 19 above)
Comment (SarahAlidoost) [reply _episodes/03-configuration.md:100]: It is shown in the correct style on the webpage. Could you please check it via [previewing your changes locally](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md#previewing-your-changes-locally)

File: _episodes/03-configuration.md (hunk 13 above)
Comment (SarahAlidoost) [reply _episodes/03-configuration.md:58]: It is shown in the correct style on the webpage. Could you please check it via [previewing your changes locally](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md#previewing-your-changes-locally)

File: _episodes/03-configuration.md (hunk 15 above)
Comment (SarahAlidoost) [reply _episodes/03-configuration.md:80]: good questions. some points for a discussion. I need to check with other episodes.

File: _episodes/03-configuration.md (hunk 18 above)
Comment (SarahAlidoost) [reply _episodes/03-configuration.md]: The instruction on how to get the data (or its rootpath) will come in setup section of the tutorial. This is the most common approach for several lessons that I have seen in software carpentry. The setup section contains everything related to data and additional material. Therefore, the episode content won't be changed. Please also see the discussion in the comment [here](https://github.com/ESMValGroup/tutorial/pull/12#discussion_r393092548)

File: _episodes/03-configuration.md (hunk 20 above)
Comment (SarahAlidoost) [reply _episodes/03-configuration.md:108]: I added more explanations.

File: _episodes/03-configuration.md (hunk 21 above)
Comment (SarahAlidoost) [reply _episodes/03-configuration.md]: I revised the text here. Please also see the e[smvaltool documentation ](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/quickstart/configure.html?highlight=parallel#user-configuration-file) where mentions that "_Set to null to use the number of available CPUs."_

File: _episodes/03-configuration.md (hunk 22 above)
Comment (SarahAlidoost) [reply _episodes/03-configuration.md]: @hb326 thank you. I will consider it for the debugging episode.

File: _episodes/03-configuration.md (hunk 28 above)
Comment (SarahAlidoost) [reply _episodes/03-configuration.md]: I revised the title. This section will be shown as a challenge. A learner needs to answer the question. The solution is also provided.

Comment (hb326) [reply _episodes/03-configuration.md]: Ok.

File: _episodes/03-configuration.md (hunk 21 above)
Comment (hb326) [reply _episodes/03-configuration.md]: OK.

File: _episodes/03-configuration.md (hunk 18 above)
Comment (hb326) [reply _episodes/03-configuration.md]: Ok.

File: _episodes/03-configuration.md (hunk 13 above)
Comment (hb326) [reply _episodes/03-configuration.md:58]: Sorry, I have trouble today with everything, as it seems. Can't get it to work. Maybe @LisaBock can have a look if it is displayed correctly?

File: _episodes/03-configuration.md [hunk 30]
```diff
@@ -3,91 +3,236 @@ title: "Configuration"
...
-~~~
-  cp config-user-example.yml config-user.yml
+The ``config-user.yml`` configuration file contains all the global level information
+needed by ESMValTool to run. This is an
+[YAML file](https://yaml.org/spec/1.2/spec.html). An example configuration file
+can be found in the root directory of the ESMValTool repository:
+[config-user-example.yml](https://github.com/ESMValGroup/ESMValTool/blob/master/config-user-example.yml).
+
+Let's download it to our working directory ``esmvaltool_tutorial``
+that was created during the [Setup]({{ page.root }}{% link setup.md %}).
```
Comment (sverhoeven) [inline _episodes/03-configuration.md:26]: The setup.md is of #46 and the installation instructions of #31 , don't create an `esmvaltool_tutorial` directory. Can the directory reason and creation be described here or in  #31 or in #46?

File: _episodes/03-configuration.md (hunk 13 above)
Comment (LisaBock) [reply _episodes/03-configuration.md:58]: Yes, looks correct.

File: _episodes/03-configuration.md [hunk 31]
```diff
@@ -3,91 +3,236 @@ title: "Configuration"
...
+To do that, click on
+[this link](https://raw.githubusercontent.com/ESMValGroup/ESMValTool/master/config-user-example.yml)
+to see a raw version of the file, right-click and press ``save as``,
+then you can rename it to ``config-user.yml``and save it into the working directory
+``esmvaltool_tutorial``.
+
+Now, let's change our working directory in a terminal window to ``esmvaltool_tutorial``.
+Then, we run a text editor called Nano to open the configuration file:
```
Comment (sverhoeven) [inline _episodes/03-configuration.md:34]: Why am I opening the config file in an editor?
```suggestion
Then, we run a text editor called Nano to have a look inside the configuration file:
```

File: _episodes/03-configuration.md (hunk 19 above)
Comment (LisaBock) [reply _episodes/03-configuration.md:100]: Format looks correct, but I would remove the three dots after 'ESMValTool' and before '[documentation]'.

File: _episodes/03-configuration.md (hunk 30 above)
Comment (SarahAlidoost) [reply _episodes/03-configuration.md]: now the directory creation is explained inside this lesson.

File: _episodes/03-configuration.md (hunk 15 above)
Comment (SarahAlidoost) [reply _episodes/03-configuration.md:91]: I added a link to the esmvaltool documentation about other projects.

**Comment by SarahAlidoost** [comment]
@sverhoeven, @LisaBock, @hb326, @bouweandela thank you for your helpful comments. I revised the text addressing your comments. It seems that some codacy issues cannot be fixed at this moment. Can we merge this pull request?

**Comment by hb326** [comment]
> 
> 
> @sverhoeven, @LisaBock, @hb326, @bouweandela thank you for your helpful comments. I revised the text addressing your comments. It seems that some codacy issues cannot be fixed at this moment. Can we merge this pull request?

Fine with me!
